internal/inventory/config.go

```diff
@@ -30,17 +26,28 @@ type ApplicationSettings struct {
 	AdjustLastReadOnByOrigin bool
 }
 
-// WriteableConfig is a struct representation of the Writeable section of the configuration.toml file.
-type WriteableConfig struct {
-	LogLevel string
+// CustomConfig is the struct representation of all of the sections from the configuration.toml
+// that we are interested in syncing with Consul.
+type CustomConfig struct {
+	AppSettings ApplicationSettings
+	Aliases     map[string]string
 }
 
-// ConsulConfig is the struct representation of all of the sections from the configuration.toml
-// that we are interested in syncing with Consul.
-type ConsulConfig struct {
-	Writable            WriteableConfig
-	ApplicationSettings ApplicationSettings
-	Aliases             map[string]string
+type ServiceConfig struct {
+	AppCustom CustomConfig
+}
+
+// UpdateFromRaw updates the service's full configuration from raw data received from
+// the Service Provider.
+func (c *ServiceConfig) UpdateFromRaw(rawConfig interface{}) bool {
+	configuration, ok := rawConfig.(*ServiceConfig)
+	if !ok {
+		return false //errors.New("unable to cast raw config to type 'ServiceConfig'")
+	}
+
+	*c = *configuration
+
+	return true
```

Comment:
Not related to this line, but a few lines down in the file, `ErrUnexpectedConfigItems` and `ErrMissingRequiredKey` are shown as unused and can be deleted.

Reply:
removed